gcc/d/dfrontend/declaration.c

```diff
@@ -1915,7 +1920,7 @@ bool VarDeclaration::checkNestedReference(Scope *sc, Loc loc)
     /* __require and __ensure will always get called directly,
      * so they never make outer functions closure.
      */
-    if (fdthis->ident == Id::require && fdthis->ident == Id::ensure)
+    if (fdthis->ident == Id::require || fdthis->ident == Id::ensure)
```

Comment:
This oops introduced in the backport of 2.071 hid a bug.

Fix has been introduced in this PR and [upstream](https://github.com/dlang/dmd/pull/7157).